File: tests/rules_lang/cc/fibonacci_test.cc
//! Fibonacci integration test

#include "fibonacci.h"

#include <cassert>
#include <vector>

int main(int argc, char** argv) {
  std::vector<std::uint64_t> numbers = {0,  1,  1,  2,  3,  5,  8,
                                        13, 21, 34, 55, 89, 144};
  for (std::size_t i = 0; i < numbers.size(); ++i) {
    assert(numbers[i] == fibonacci::fibonacci(i));
  }

  return 0;
}

File: tests/rules_lang/rust/fibonacci_test.rs
//! Fibonacci integration test

use fibonacci::fibonacci;

#[test]
fn test_fibonacci() {
    let numbers: Vec<u64> = vec![0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144];
    for (i, number) in numbers.iter().enumerate() {
        assert_eq!(*number, fibonacci(i as u64));
    }
}
